notifique/src/main/java/com/nathanrassi/notifique/NotifiqueActivity.kt

```diff
@@ -29,33 +31,20 @@ class NotifiqueActivity : AppCompatActivity() {
     checkNotificationPermission()
   }
 
-  override fun onActivityResult(
-    requestCode: Int,
-    resultCode: Int,
-    data: Intent?
-  ) {
-    super.onActivityResult(requestCode, resultCode, data)
-    if (requestCode == notificationPermissionRequestCode) {
-      checkNotificationPermission()
-    }
-  }
-
   private fun checkNotificationPermission() {
     if (!NotificationManagerCompat.getEnabledListenerPackages(this).contains(packageName)) {
+
       val notificationSnackbar = Snackbar.make(
           findViewById(android.R.id.content),
           R.string.snackbar,
           Snackbar.LENGTH_INDEFINITE
       )
-      notificationSnackbar.setAction(R.string.snackbar_action) {
-        val intent = Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS)
-        if (packageManager.queryIntentActivities(intent, 0).isEmpty()) {
-          // TODO: rare problem.
-        } else {
-          startActivityForResult(intent, 0)
+      class SnackbarNotificationListener : View.OnClickListener {
+        override fun onClick(v: View) {
+          startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS))
```

Comment:
let's leave the packageManager check from above. some devices wont have the same settings screen. im not sure it's guaranteed or tested in the Compatibility Test Suite.

Reply:
sounds good. I can change that